tests/unit/sagemaker/image_uris/test_autogluon.py

```diff
@@ -42,30 +42,53 @@
     "us-west-1": "763104351884",
     "us-west-2": "763104351884",
 }
-VERSIONS = ["0.3.1", "0.3.2", "0.4.0", "0.3", "0.4"]
+VERSIONS = ["0.3.1", "0.3.2", "0.4.0", "0.4.2", "0.3", "0.4"]
+SCOPES = ["training", "inference"]
+PROCESSORS = ["cpu", "gpu"]
 
 
 @pytest.mark.parametrize("version", VERSIONS)
-def test_valid_uris(version):
+@pytest.mark.parametrize("scope", SCOPES)
+@pytest.mark.parametrize("processor", PROCESSORS)
+def test_valid_uris_training(version, scope, processor):
+    instance_type = "ml.c4.xlarge" if processor == "cpu" else "ml.p2.xlarge"
     py_version = "py37" if version == "0.3.1" else "py38"
-    for region in ACCOUNTS.keys():
-        uri = image_uris.retrieve(
-            "autogluon",
-            region=region,
-            version=version,
-            py_version=py_version,
-            image_scope="training",
-            instance_type="ml.c4.xlarge",
-        )
+    if (
+        scope == "inference"
+        and processor == "gpu"
+        and version in ["0.3.1", "0.3.2", "0.4.0", "0.3"]
```

Comment:
do we need "0.4.2" as well here with "0.4.0

Reply:
No, `0.4.2` has GPU container, while earlier releases don't.